remote-trait-object/src/service/remote.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::service::{Handle, MethodId};
+use crate::Packet;
+
+impl Handle {
+    pub fn call<S: serde::Serialize, D: serde::de::DeserializeOwned>(&self, method: MethodId, args: &S) -> D {
```

Comment:
How about `send_message_to_remote`, `call_to_remote`, `call_remote_method`, ...?
It's hard to catch the concept with the name `call` in the `Handle` struct.

Reply:
@majecty I think just `call` can be quite consistent. For example, the `callslots`. If you want to emphasize the remoteness, if is better to clarify the role of `Handle`, instead sticking to `call`, since `call` is obviously part of the `Handle`.